### PR TITLE
[FFM-926]: updating code to make metrics service tracking a non blocking call

### PR DESCRIPTION
**Pull request by rushabh-harness**
Updating the blocking call in ringbuffer getNext to tryNext. 

Removing unwanted logs, adding a debug option in CfClient.

### CODE REVIEW

File: src/main/java/io/harness/cf/client/api/analytics/AnalyticsManager.java [hunk 1]
```diff
@@ -66,15 +69,19 @@ public void pushToQueue(Target target, FeatureConfig featureConfig, Variation va
...
-    long sequence = ringBuffer.next(); // Grab the next sequence
+    long sequence = -1;
     try {
+      sequence = ringBuffer.tryNext(); // Grab the next sequence
       Analytics event = ringBuffer.get(sequence); // Get the entry in the Disruptor for the sequence
       event.setFeatureConfig(analytics.getFeatureConfig());
       event.setTarget(analytics.getTarget());
       event.setVariation(analytics.getVariation());
+    } catch (InsufficientCapacityException e) {
+      log.debug("Insufficient capacity in the analytics ringBuffer", e);
```
Comment (davejohnston) [inline src/main/java/io/harness/cf/client/api/analytics/AnalyticsManager.java:80]: I wonder if this should be a warn level?

File: src/main/java/io/harness/cf/client/api/analytics/TimerTask.java [hunk 2]
```diff
@@ -16,13 +17,18 @@ public TimerTask(RingBuffer<Analytics> ringBuffer) {
...
-    long sequence = ringBuffer.next(); // Grab the next sequence
+    long sequence = -1;
     try {
-      log.info("Publishing timerInfo to ringBuffer");
+      sequence = ringBuffer.tryNext(); // Grab the next sequence if we can
+      log.debug("Publishing timerInfo to ringBuffer");
       Analytics event = ringBuffer.get(sequence); // Get the entry in the Disruptor for the sequence
       event.setEventType(EventType.TIMER);
+    } catch (InsufficientCapacityException e) {
+      log.debug("Insufficient capacity in the analytics ringBuffer", e);
```
Comment (davejohnston) [inline src/main/java/io/harness/cf/client/api/analytics/TimerTask.java:27]: Same as the other one - it may make sense to be warn

**Review by davejohnston — APPROVED** [review]
LGTM.  I left some comments around log level but you can decide if it makes sense or not.

File: src/main/java/io/harness/cf/client/api/analytics/AnalyticsManager.java (hunk 1 above)
Comment (rushabh-harness) [reply src/main/java/io/harness/cf/client/api/analytics/AnalyticsManager.java:80]: ok, I ll remove the stacktrace then

File: src/main/java/io/harness/cf/client/api/analytics/AnalyticsCacheFactory.java [hunk 3]
```diff
@@ -9,11 +9,11 @@
...
   public static Cache create(String cacheName) {
     if (cacheName.equals(CAFFEINE_CACHE)) {
-      log.info("Using Caffeine cache");
+      log.debug("Using Caffeine cache");
       return new CaffeineCache();
     }
     if (cacheName.equals(GUAVA_CACHE)) {
-      log.info("Using Guava cache");
+      log.debug("Using Guava cache");
       return new GuavaCache();
```
Comment (enver-bisevac) [inline src/main/java/io/harness/cf/client/api/analytics/AnalyticsCacheFactory.java:17]: I think it is better to have default cache and if customer want to change cache implementation just need to specify dependency in client builder.

**Pull request merged**